### PR TITLE
feat(hubspot): add HubSpot CRM integration with SDK 2.0.0

### LINKED ISSUE #263: feat(hubspot): migrate from private repo, upgrade to SDK 2.0.0, add pytest unit + integration tests
## Summary

Migrate the HubSpot CRM integration from the private `integrations` repo to the public `autohive-integrations` repo.

## Scope

- **48 actions** across 3,442 lines of Python and 4,622 lines of config.json
- Domains: Contacts, Companies, Deals, Tickets, Notes, Lists, Associations, Marketing Emails, Campaigns, Calls/Meetings with Transcripts, Owners

## Work Items

1. **Copy integration** to public repo (hubspot.py, config.json, requirements.txt, icon.png, README.md)
2. **Upgrade to SDK 2.0.0**:
   - `context.fetch()` returns `FetchResponse` — add `.data` to all call sites (~80+)
   - `parse_response()` rewrite — remove `hasattr(response, "json")` pattern
   - Convert all error returns to `ActionError(message=...)` (~20+ catch blocks)
   - Bump `requirements.txt` to `~=2.0.0` and `config.json` version to `2.0.0`
3. **Write pytest unit tests** (`test_hubspot_unit.py`) with `mock_context` + `FetchResponse` pattern — targeting broad coverage across all 48 actions
4. **Convert old live tests** to `test_hubspot_integration.py` with `@pytest.mark.integration` markers, env-based credentials, and proper assertions

## Notes

- Existing tests in private repo are manual smoke tests (no mocking, no assertions, hardcoded placeholders) — not suitable for CI
- `create_contact` uses legacy v1 API (`/contacts/v1/contact`) — flagged for future cleanup
- No secrets or internal URLs in codebase — safe for public migration

### PR DESCRIPTION
**Pull request by TheRealAgentK**
## Summary

Migrates the HubSpot CRM integration from the private `integrations` repo with SDK 2.0.0 upgrade and comprehensive test coverage.

Closes #263

### Integration (48 actions)
- **Contacts**: get, create, update, search, recent, emails, add to list
- **Companies**: get, create, update, search, search by owner name
- **Deals**: get, create, update, search, recent, pipelines, fetch_all pagination
- **Tickets**: recent, conversation threading, comments
- **Notes**: create, update, delete (with multi-object associations)
- **Lists**: get, search, memberships, members with batch contact fetch
- **Associations**: contact/company/deal associations (v4 API)
- **Marketing**: emails, campaigns, campaign assets, campaign performance
- **Calls/Meetings**: with transcript retrieval (Calling Transcripts API)
- **Properties**: contact, company, deal property discovery
- **Owner**: get by ID

### SDK 2.0.0 Upgrade
- `context.fetch()` → `FetchResponse.data` pattern throughout
- `parse_response()` simplified to `return response.data`
- All error returns converted to `ActionError(message=...)`
- `requirements.txt` → `~=2.0.0`, `config.json` version → `2.0.0`

### Unit Tests (264 tests across 7 files)
Split by domain for maintainability:

| File | Tests | Coverage |
|---|---|---|
| `test_hubspot_helpers_unit.py` | 32 | parse_response, date parsing, timestamp conversion |
| `test_hubspot_contacts_unit.py` | 38 | 8 contact actions |
| `test_hubspot_companies_unit.py` | 47 | 10 company/properties actions |
| `test_hubspot_deals_unit.py` | 38 | 11 deal/calls/transcript actions |
| `test_hubspot_notes_unit.py` | 27 | 3 note CRUD actions |
| `test_hubspot_tickets_unit.py` | 25 | 3 ticket actions |
| `test_hubspot_misc_unit.py` | 57 | lists, list members, associations, marketing, owner |

Each action tested with: happy path, request verification, error handling, edge cases, response shape.

### Integration Tests (41 tests)
- 30 read-only tests (safe against any HubSpot account)
- 11 destructive tests marked `@pytest.mark.destructive` (create/update/delete operations)
- Note lifecycle workflow: create → read → update → delete
- All credentials via env vars with graceful skip when not set

### Other Changes
- HubSpot entry added to main `README.md`
- HubSpot env vars added to `.env.example`
- `destructive` pytest marker registered in `pyproject.toml`

### Bugs fixed during review
- **`get_ticket_conversation`**: no-thread path returned `{"result": ...}` instead of `{"conversation": ...}`, causing a validation error. Now returns consistent conversation-shaped response with empty results.
- **`get_recent_contacts`**: code read `count` from inputs but schema declares `limit`. Now reads `limit`.
- **Integration tests**: `TestGetDeals`/`TestSearchDeals` asserted `"deals"` key but actions return `"results"`. Fixed.
- **Integration tests**: `TestGetContact` used hardcoded `test@example.com` with `assert result is not None`. Now uses `HUBSPOT_TEST_CONTACT_EMAIL` env var with structural assertions.

## Known issues (tracked separately)
- Property actions (`get_company_properties`, `get_deal_properties`, `get_contact_properties`) return strings when `include_details=false` but `get_company_properties` schema declares object items — tracked in #265
- Unused scopes: `oauth`, `content` — used by HubSpot OAuth flow
- `create_contact` uses legacy v1 API — flagged for future cleanup

### CODE REVIEW

File: hubspot/hubspot.py [hunk 1]
```diff
@@ -0,0 +1,3321 @@
...
+        # Retrieve the correct thread ID using the helper function.
+        thread_id = await get_thread_id_from_ticket(ticket_id, context)
+        if not thread_id:
+            return ActionResult(
+                data={
+                    "result": {
+                        "success": False,
+                        "message": f"No conversation thread found for ticket {ticket_id}",
+                    }
+                },
```
[the remark this answers is not in the document]
Comment (TheRealAgentK) [reply hubspot/hubspot.py]: Addressed https://github.com/Autohive-AI/autohive-integrations/pull/264/commits/8b847a57befb253ed120d677d0cd2dc83a3e6761

File: hubspot/hubspot.py [hunk 2]
```diff
@@ -0,0 +1,3321 @@
...
+                    "groupName": prop.get("groupName"),
+                    "hubspotDefined": prop.get("hubspotDefined", True),
+                    "description": prop.get("description"),
+                    "options": prop.get("options", []),
+                }
+                for prop in results
+            ]
+        else:
+            # Return just property names (backward compatible)
+            properties = [prop.get("name") for prop in results if prop.get("name")]
```
[the remark this answers is not in the document]
Comment (TheRealAgentK) [reply hubspot/hubspot.py:1327]: Tracked for future consideration in #265

File: hubspot/hubspot.py [hunk 3]
```diff
@@ -0,0 +1,3321 @@
...
+        """
+        Execute the get_recent_contacts action.
+
+        :param inputs: Dictionary with key "count" specifying the maximum number of contacts.
+        :param context: Execution context containing authentication and fetch method.
+        :return: Dictionary with a "recent_contacts" key containing recent contact data.
+        """
+
+        count = inputs.get("count", 100)
+        url = f"https://api.hubapi.com/crm/v3/objects/contacts?limit={count}&sort=createdat"
```
[the remark this answers is not in the document]
Comment (TheRealAgentK) [reply hubspot/hubspot.py]: Correct, addresed in https://github.com/Autohive-AI/autohive-integrations/pull/264/commits/b8e2c91169677e69926a69a872e7ef028f8e91fe

**Comment by TheRealAgentK** [comment]
All review comments from Codex have been addressed.

**Review by NinosMan — APPROVED** [review]
